src/systems/render-manager/hubs-batch-raw-uniform-group.js

```diff
@@ -60,7 +69,11 @@ export default class HubsBatchRawUniformGroup extends BatchRawUniformGroup {
       // TODO need to account for nested visibility deeper than 1 level
       this.setInstanceTransform(
         instanceId,
-        mesh.visible && (mesh.parent && mesh.parent.visible) ? mesh.matrixWorld : HIDE_MATRIX
+        mesh.visible &&
+        (mesh.parent && mesh.parent.visible) &&
+        (!cameraSystem.inspected || inParentHierarchyOf(cameraSystem.inspected, mesh))
```

Comment:
this is a hotspot, so i would push `!!cameraSystem.inspected` into a `const` boolean variable above to ensure this expression doesn't need to deref `cameraSystem` on every iteration and also have a mixture of types in the expression.

you also might want to just consider splitting it into two branches, to avoid evaluating the ternary operator on every instance.

Reply:
:+1:

Reply:
Ok I'll pull the dereference out of the loop, and I can find which `mesh` in the batch will be affected once when transitioning to inspect mode and keep that state around, rather than checking `inParentHierarchyOf` for every object in the batch each frame.